### PR TITLE
Adds Humanoid, Health, MaxHealth, Died

### PR DESCRIPTION
**Pull request by Kampfkarren**

### CODE REVIEW

File: lib/instances/Humanoid.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		validateType("Health", value, "number")
+		local instance = getmetatable(self).instance
+		local health = math.min(
+			math.max(0, value),
+			self.MaxHealth
+		)
+
+		instance.properties.Health = health
+
+		if not instance.died and health == 0 and self:FindFirstAncestorWhichIsA("Workspace") then
```
Comment (LPGhatguy) [inline lib/instances/Humanoid.lua]: Does a `Humanoid` with 0 health that gets parented to `Workspace` trigger `Died` as well?

Comment (LPGhatguy) [inline lib/instances/Humanoid.lua]: Can we explicitly compare instances with `nil` in cases like this, too?

File: lib/instances/Humanoid.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	end,
+})
+
+Humanoid.properties.MaxHealth = InstanceProperty.normal({
+	getDefault = function()
+		return 100
+	end,
+
+	set = function(self, key, value)
+		validateType("Health", value, "number")
```
Comment (LPGhatguy) [inline lib/instances/Humanoid.lua]: This is `MaxHealth`, not `Health`.

File: lib/instances/Humanoid.lua (hunk 1 above)
Comment (Kampfkarren) [reply lib/instances/Humanoid.lua]: No.

**Comment by Kampfkarren** [comment]
@LPGhatguy Fixed

File: lib/instances/Humanoid_spec.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
+local Humanoid = import("./Humanoid")
+local Workspace = import("./Workspace")
+
+describe("instances.Humanoid", function()
+	it("should instantiate", function()
+		local instance = Humanoid:new()
```
Comment (LPGhatguy) [inline lib/instances/Humanoid_spec.lua]:
```suggestion
		local instance = Instance.new("Humanoid")
```

Part of this test's utility is making sure that Humanoid is actually creatable. You'll also need to add an import for `Instance` to this file.

Comment (Kampfkarren) [reply lib/instances/Humanoid_spec.lua]: Just as I was about to say how cool GitHub suggested changes are, I see that you couldn't just write the import yourself with it :(

(Fixing it)

Comment (Kampfkarren) [reply lib/instances/Humanoid_spec.lua]: Fixed.

**Comment by LPGhatguy** [comment]
Just waiting on Travis!

**Pull request merged**